python/ray/data/tests/test_execution_optimizer.py

```diff
@@ -1151,6 +1151,30 @@ def test_sort_validate_keys(ray_start_regular_shared):
         ds_named.sort(invalid_col_name).take_all()
 
 
+def test_sort_on_group_data(ray_start_regular_shared):
```

Comment:
could you also add similar tests to cover other related all-to-all ops (shuffle, aggregate)? thanks!

Reply:
Just added one for aggregate. the `RandomShuffle` requires more changes with `Repartition` as well. so I just update tests for `Sort` and `Aggregate`.